### ISSUE TITLE
in autotools build, --disable-remote-functionality does not turn off nczarr

### ISSUE THREAD

**Issue by edwardhartnett** [Unidata/netcdf-c#3021]
./configure --disable-netcdf-4 --prefix=/usr/local/netcdf-c-classic --disable-remote-functionality

yields:

```
# NetCDF C Configuration Summary
==============================

# General
-------
NetCDF Version:		4.9.4-development
Dispatch Version:       5
Configured On:		Thu Sep 12 07:15:41 MDT 2024
Host System:		x86_64-pc-linux-gnu
Build Directory: 	/home/ed/netcdf-c
Install Prefix:         /usr/local/netcdf-c-classic
Plugin Install Prefix:  /usr/local/hdf5/lib/plugin

# Compiling Options
-----------------
C Compiler:		/usr/bin/gcc
CFLAGS:			 -fno-strict-aliasing
CPPFLAGS:		
LDFLAGS:		
AM_CFLAGS:		
AM_CPPFLAGS:		
AM_LDFLAGS:		
Shared Library:		yes
Static Library:		yes
Extra libraries:	-lm -lz -lsz -lzstd -lxml2 -lcurl 
XML Parser:             libxml2

# Features
--------
Benchmarks:		no
NetCDF-2 API:		yes
HDF4 Support:		no
HDF5 Support:		no
CDF5 Support:		yes
NC-4 Parallel Support:	no
PnetCDF Support:	no

DAP2 Support:		no
DAP4 Support:		no
Byte-Range Support:	no

S3 Support:	        no
S3 SDK:  	        none

NCZarr Support:		yes
NCZarr Zip Support:     no

Diskless Support:	yes
MMap Support:		no
ERANGE Fill Support:	no
Relaxed Boundary Check:	yes

Plugins Enabled:        yes
Plugin Install Dir:     /usr/local/hdf5/lib/plugin

Quantization:		yes
Logging:     		no
SZIP Write Support:     yes
Standard Filters:       bz2 deflate szip zstd
ZSTD Support:           yes
Parallel Filters:       no

```

Note that nczarr is enabled, but should not be.

Also, should plugins be enabled in this case? Without zarr and hdf5, there is no reason for plugins.

**Comment by dopplershift** [comment]
nczarr isn't strictly a remote storage format. For instance, the netCDF library supports doing zarr locally as a file tree or within a zip file, so in my mind it makes sense for it to remain enabled when building without remote functionality.

**Comment by WardF** [comment]
Agreed.

**Comment by DennisHeimbigner** [comment]
Ditto. disabling remote functionality does need to turn off S3 support in NCzarr.

**Comment by edwardhartnett** [comment]
OK, I accepted this at first, but on further consideration I think it's the incorrect answer.

The point of --disable-remote-functionality is to make netCDF safer on high-FISMA systems. That means that when sysadmins use --disable-remote-functionality, *all* remote functionality should be disabled (and not even built into the library), and curl should not be linked.

The fact that nczarr has a local-only possibility is not the point - it's remote capabilities are the point. So if you can build a local nczarr with no link to curl, then that's OK with --disable-remote-functionality, but if nczarr cannot be built without any internet connectivity code, then nczarr should also be turned off with --disable-remote-functionality.

In other words, --disable-remote-functionality should decisively and completely disable all remote functionality. If it does not do that, than just take it out, because it's misleading. It's only purpose is to support high-FISMA systems.

**Comment by dopplershift** [comment]
Thinking that way, I agree with @edwardhartnett. Otherwise, the option should be `--disable-some-remote-functionality` 😉 In my head, it was implicit that you could have a local-only build of nczarr support, but I realize that might be a bigger lift.

**Comment by DennisHeimbigner** [comment]
You might want to rethink that. You appear to be saying that any file format (dispatcher) should be disabled if any subpart of it can do remote access. But byterange access is currently provided in both netcdf-3 and in HDF5.
That would seem to imply that they need to be completely disabled.

**Comment by dopplershift** [comment]
I think a careful consideration of the audience/use-case that originally motivated `--disable-remote-functionality`, plus what would be least surprising to a user first trying it, is needed. 

My personal expectation is that if I pass `--disable-remote-functionality` to some piece of software, it should no longer be able to do any remote access. That said, I'll defer that to y'all and the audience of users who need that flag, because it's not something I would plan on using either.

**Comment by edwardhartnett** [comment]
@DennisHeimbigner WRT byterange, it should also be turned off by disable-remote-functionality.

I'm not saying any dispatcher which does remote access should not be built, if they can be built without remote functionality. When we build classic with -disable-byterange, that's what we get, right?

**Comment by DennisHeimbigner** [comment]
Correct. But I interpreted a prior message as asking to completelt disablig NCZarr as opposed to just disabling byterange and S3.

**Comment by edwardhartnett** [comment]
If disabling byterange and s3 causes all remote behavior to be disabled, that's all we need.
